keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-	PUBLIC_READ_WRITE = 3;
-    }
-    
-    AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
+```yaml
+BucketClass {
+  TypeMeta
+  ObjectMeta
+  
+  // Provisioner is the name of driver associated with this bucket
+  Provisioner string
+  
+  // DeletionPolicy is used to specify how COSI should handle deletion of this 
+  // bucket. There are 3 possible values:
+  //  - Retain: Indicates that the bucket should not be deleted from the OSP
+  //  - Delete: Indicates that the bucket should be deleted from the OSP
+  //        once all the workloads accessing this bucket are done
+  //  - ForceDelete: Indicates that the bucket and its contents should be 
+  //        deleted from the OSP even if workloads are currently accessing 
+  //        this bucket
+  DeletionPolicy DeletionPolicy
+
+  // AllowedNamespaces is the list of namespaces from which workloads     
+  // can access this Bucket. If empty, all namespaces can access this access
+  // +optional   
+  AllowedNamespaces []string
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string 
 }
+```
 
-message ProvisionerCreateBucketResponse {
-    // Intentionally left blank
+## BucketAccess
+
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketAccessSpec {
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating the bucket
+    // +optional
+    BucketAccessRequest corev1.ObjectReference
+    
+    // Name of the BucketAccessClass specified in the BucketAccessRequest
+    BucketAccessClassName string
+    
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating access credentials
+    // +optional
+    Parameters map[string]string
+  }
+  
+  Status BucketAccessStatus{
+    // AccessGranted indicates that the credentials provisioned have been
+    // granted access to the bucket
+    AccessGranted bool
+  
+    // AccountID is the unique ID for the account in the OSP
+    AccountID string
+  
+    // CredentialsSecret is the secret which holds information the
+    // accessKey and secretKey values of the granted credentials
+    CredentialsSecret corev1.ObjectReference
+  }
 }
 ```
 
-## ProvisonerDeleteBucket
-
-This call is made to delete the bucket in the backend. If the bucket has already been deleted, then no error should be returned. The call to *ProvisonerDeleteBucket* MUST be idempotent.
+## BucketAccessRequest
 
-```
-message ProvisionerDeleteBucketRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
+```yaml
+BucketAcessRequest {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketAccessRequestSpec {
+    // BucketRequestName is the name of the BucketRequest
+    BucketRequestName string
     
-    map<string,string> bucket_context = 2;    
-}
+    // BucketAccessClassName is the name of the BucketAccessClass
+    // +optional    
+    BucketAccessClassName string
+  }
```

Comment:
This spec requires a BR to be referenced for every BAR, which also implies that a BC is needed for every BAR. This differs from the earlier KEP. How is bucket sharing across namespaces handled?

Reply:
BAR.spec.serviceAccountName was removed which is fine. Is the idea now to include the SA in `parameters` if if the OSP supports k8s SA intergration? I think COSI needs to help enable SA integration to object store systems.

Reply:
> BAR.spec.serviceAccountName was removed, which is fine if the idea is to include the SA in parameters if the OSP supports k8s SA intergration. I think COSI needs to help enable SA integration in object store systems.

Yes

Reply:
> This spec requires a BR to be referenced by every BAR, which also implies that a BC is needed for every BAR. This differs from the earlier KEP where the BR name was optional and a B.name could be supplied. How is bucket sharing across namespaces handled?

There was a mistake while copying fields over. There is a reference to Bucket in the BA, and references to both BAR and B in the BAR.